lib/webauth.js

```diff
@@ -140,15 +146,15 @@ module.exports = function(authManager) {
       handler.addProxyAuthorizations(req1, req2Options, authPluginSession);     
     },
     
-    getStatus(req, res) {
+    getStatus: Promise.coroutine(function*(req, res) {
       const handlers = authManager.getAllHandlers();
       const result = new StatusResponse();
       for (const handler of handlers) {
         const pluginID = handler.pluginID;
         const authPluginSession = util.getOrInit(req.session, pluginID, {});
         let status;
         try {
-          status = handler.getStatus(authPluginSession); 
+          status = yield handler.getStatus(req, authPluginSession); 
```

Comment:
So the status is updated both implicitly (when the client calls getStatus()) and explicitly when the client calls refreshStatus()? What should be the difference for the client? Why won't it, let's say, call getStatus() in both cases, if it knows that getStatus() would also update thing anyway?

However, I initially thought of getStatus() as a pure "getter" function that doesn't change any state. Why is the implicit update necessary? Can the client simply explicitly call refreshStatus() when needed?

Reply:
OK, we'll make getStatus() not be usable for refresh (as in, it wont send back any new expiration info as its not intended to renew or update expiration), and keep refresh as-is.